plugins/souin/types/layerStorage.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	"github.com/dgraph-io/ristretto"
+)
+
+// CoalescingLayerStorage is the layer to be able to not coalesce uncoalesceable request
+type CoalescingLayerStorage struct {
+	*ristretto.Cache
+}
+
+// InitializeCoalescingLayerStorage initialize the storage
+func InitializeCoalescingLayerStorage() *CoalescingLayerStorage {
+	storage, _ := ristretto.NewCache(&ristretto.Config{
+		NumCounters: 1e7,     // number of keys to track frequency of (10M).
+		MaxCost:     1 << 30, // maximum cost of cache (1GB).
+		BufferItems: 64,      // number of keys per Get buffer.
+	})
+
+	return &CoalescingLayerStorage{Cache: storage}
+}
+
+// Exists method returns if the key should coalesce
+func (provider *CoalescingLayerStorage) Exists(key string) bool {
+	_, found := provider.Cache.Get(key)
+	return !found
+}
+
+// Set method will store the response in Ristretto provider
```

Comment:
`// Set method will store the response in Ristretto provider`
Will store "the response"? Only `nil` is stored here…

Reply:
that's unused in the Souin codebase, I don't know why I pushed that.